packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.m

```diff
@@ -108,6 +109,134 @@ - (void)testGetProductResponse {
+      [[SKPaymentTransactionStub alloc] initWithMap:transactionMap];
+  NSArray *array = [NSArray arrayWithObjects:paymentTransaction, nil];
+
+  FIAPaymentQueueHandler *mockHandler = OCMClassMock(FIAPaymentQueueHandler.class);
+  OCMStub([mockHandler getUnfinishedTransactions]).andReturn(array);
+
+  self.plugin.paymentQueueHandler = mockHandler;
+
+  FlutterError *error;
+  [self.plugin finishTransactionFinishMap:args error:&error];
+
+  XCTAssertNil(error);
+}
+
+- (void)testFinishTransactionSucceedsWithNilTransaction {
+  NSDictionary *args = @{
+    @"transactionIdentifier" : [NSNull null],
+    @"productIdentifier" : @"unique_identifier",
+  };
+
+  NSDictionary *paymentMap = @{
+    @"productIdentifier" : @"123",
+    @"requestData" : @"abcdefghabcdefghabcdefghabcdefghabcdefghabcdefghabcdefghabcdefgh",
+    @"quantity" : @(2),
+    @"applicationUsername" : @"app user name",
+    @"simulatesAskToBuyInSandbox" : @(NO)
+  };
+
+  NSDictionary *transactionMap = @{
+    @"transactionState" : @(SKPaymentTransactionStatePurchasing),
+    @"payment" : paymentMap,
+    @"error" : [FIAObjectTranslator getMapFromNSError:[NSError errorWithDomain:@"test_stub"
+                                                                          code:123
+                                                                      userInfo:@{}]],
+    @"transactionTimeStamp" : @([NSDate date].timeIntervalSince1970),
+  };
+
+  SKPaymentTransactionStub *paymentTransaction =
+      [[SKPaymentTransactionStub alloc] initWithMap:transactionMap];
+
+  FIAPaymentQueueHandler *mockHandler = OCMClassMock(FIAPaymentQueueHandler.class);
+  OCMStub([mockHandler getUnfinishedTransactions]).andReturn(@[ paymentTransaction ]);
+
+  self.plugin.paymentQueueHandler = mockHandler;
+
+  FlutterError *error;
+  [self.plugin finishTransactionFinishMap:args error:&error];
+
+  XCTAssertNil(error);
+}
+
+- (void)testGetProductResponseWithRequestError {
+  NSArray *argument = @[ @"123" ];
+
+  id mockHandler = OCMClassMock([FIAPRequestHandler class]);
+  InAppPurchasePlugin *plugin = [[InAppPurchasePlugin alloc]
+      initWithReceiptManager:nil
+              handlerFactory:^FIAPRequestHandler *(SKRequest *request) {
+                return mockHandler;
+              }];
+
+  NSError *error = [NSError errorWithDomain:@"errorDomain"
+                                       code:0
+                                   userInfo:@{NSLocalizedDescriptionKey : @"description"}];
+
+  OCMStub([mockHandler
+      startProductRequestWithCompletionHandler:([OCMArg invokeBlockWithArgs:[NSNull null], error,
+                                                                            nil])]);
+
+  [plugin
+      startProductRequestProductIdentifiers:argument
+                                 completion:^(SKProductsResponseMessage *_Nullable response,
+                                              FlutterError *_Nullable startProductRequestError) {
+                                   XCTAssertNotNil(error);
```

Comment:
You'll want an expectation here to make sure that this is actually called.

Reply:
doneee